content/scalability/docs/control-plane.md

```diff
@@ -240,4 +244,50 @@ spec:
+
+## Use Shared Informers for Custom Controllers
+
+When building controllers and automation that integrate with the Kubernetes API you will often need to get information from Kubernetes resources. If you poll for these resources regularly it can cause a significant load on the API server.
+
+Using an [informer](https://pkg.go.dev/k8s.io/client-go/informers) from the client-go library will give you benefits of watching for changes to the resources based on events instead of polling for changes. Informers further reduce the load by using shared cache for the events and changes so multiple controllers watching the same resources do not add additional load.
+
+Polling resources without including labels or namespace scope is especially difficult in large clusters. Each un-filtered poll can require a lot of unnecessary data to be sent from etcd through the API server to be filtered in your controller. By filtering based on labels and namespaces you can reduce the amount of work the API server needs to perform to fullfil the request and data sent to the client.
+
+## Optimize Kubernetes API usage
+
+When calling the Kubernetes API with custom controllers or automation it's important that you limit the calls to only the resources you need. Without limits you can cause unneeded load on the API server and etcd.
+
+It is recommended that you use the watch argument whenever possible. With no arguments the default behavior is to list objects. To use watch instead of list you can append `?watch=true` to the end of your API request. For example, to get all pods with a watch use:
+
+```
+/api/v1/pods?watch=true
+```
+
+If you are listing objects you should limit the scope of what you are listing. The `fieldSelector` argument and `/namespace/` path can be useful to make sure your lists are as narrowly scoped as needed. For example, to list only running pods in the default namespace use:
+
+```
+/api/v1/namespaces/default/pods?fieldSelector=status.phase=Running
+```
+
+Or list all pods that are running with:
+
+```
+/api/v1/pods?fieldSelector=status.phase=Running
+```
+
+You should use a resourceVersion and pagination to further limit lists and reduce load. You can read about [resourceVersions in the Kubernetes documentation](https://kubernetes.io/docs/reference/using-api/api-concepts/#resource-versions). Without a resourceVersion argument you will receive the most recent version available which requires an etcd quorum read which is the most expensive and slowest read for the database. The resourceVersion depends on what resources you are trying to query and can be found in the `metadata.resourseVersion` field.
+
+You should limit your page returns to no more than 500 by appending `limit=500` to your API queries. To list 500 pods that are newer than version 1000 you can use:
+
+```
+/api/v1/pods/?limit=500&resourceVersionMatch=NotOlderThan&resourceVersion=1000
```

Comment:
The problem with specifying resourceVersion for List calls is knowing what value to use. Once you set the RV in List call with `resourceVersionMatch=NotOlderThan` API server is going to return all object at that RV and it might be stale data or a future RV. So its hard to set the RV for List calls.

Reply:
The more I read the docs on this it seems like a scaling safeguard that is difficult to practically implement.

If we don't set a limit then `NotOlderThan` is the default behavior, but if we set `limit` without `resourceVersionMatch` then exact is the default.

I'm fine leaving this section out because I'm not sure how useful it is, but if you think it's still important to include then I'm open to suggestions on how to clarify when it would be practical to use.